src/solve/execute-route.ts

```diff
@@ -0,0 +1,308 @@
+import {
+  enableChains,
+  getAddressForChain,
+  getChainByID,
+  getOfflineSigner,
+  getOfflineSignerOnlyAmino,
+  getSigningCosmWasmClientForChainID,
+  getSigningStargateClientForChainID,
+  getStargateClientForChainID,
+  isLedger,
+  signAmino,
+  signAndBroadcastEvmos,
+  signAndBroadcastInjective,
+} from "@/utils/utils";
+import { EncodeObject, OfflineSigner, coin } from "@cosmjs/proto-signing";
+import { GasPrice, DeliverTxResponse } from "@cosmjs/stargate";
+import { MsgTransfer as MsgTransferInjective } from "@injectivelabs/sdk-ts";
+import { WalletClient } from "@cosmos-kit/core";
+import { SkipClient, MsgsRequest, RouteResponse } from "./client";
+import Long from "long";
+import { OfflineAminoSigner } from "@keplr-wallet/types";
+import { TxRaw } from "cosmjs-types/cosmos/tx/v1beta1/tx";
+
+const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export async function executeRoute(
```

Comment:
This is a pretty massive function. It might be worth using a couple of helpers throughout to shorten it, so it's easier to understand what it's doing (and to de-dupe logic where possible)